### ISSUE TITLE
Remove usage of default directory for mixed refs builds

### ISSUE THREAD

**Issue by sharon-wang** [eclipse/openj9#11768]
In a regular build, sources are generated into the `default` directory for non-compressed refs build and the `compressedrefs` directory for a compressed refs build.

In mixed builds, we currently generate the sources/libs into the `default` directory. However, it's not actually necessary for a directory to be used at all for a mixed build, since sources/libraries don't need to be kept separate. So, the generated libraries can just sit in the parent directory where the `compressedrefs`/`default` directories currently sit alongside all the other libraries, i.e. `images/jdk/lib`, `jdk/lib`.

More mixed builds discussion: #8878.

**Comment by gacholio** [comment]
I'm not sure we want to do this - we had originally decided that default would be the directory for mixed builds.

**Comment by sharon-wang** [comment]
Right, using `default` keeps the JVM libraries contained and separated from the rest of the libraries. Is this separation something we want to keep going forward? If so, I can close this issue.

I think @keithc-ca might have some thoughts on this too?

**Comment by gacholio** [comment]
The plan was to allow the redirector to continue to select other VMs if we have a future experimental build that we want to package in. This doesn't necessarily preclude moving the default VM up a level, but I'm not sure there's any real value to it.

**Comment by gacholio** [comment]
@tajila

**Comment by sharon-wang** [comment]
If no other comments, I think I'll just close this issue.

**Comment by keithc-ca** [comment]
It might make a (small?) performance difference removing an extra directory from each of these search paths:
```
com.ibm.oti.vm.bootstrap.library.path
java.library.path
sun.boot.library.path
```
It takes longer to search a directory and fail than it does to succeed.

We don't *need* to eliminate the redirector, so those experiments would still be possible.
